tests/ui/type-alias-impl-trait/hidden_behind_projection_behind_struct_field.rs

```diff
@@ -0,0 +1,28 @@
+//! This test shows that a field type that is a projection that resolves to an opaque,
```

Comment:
I'm personally okay with this behavior. Seems really niche that you'd want to define a TAIT via a normalized assoc ty, seems forwards compatible to fix this later with normalization and a recursion depth or something.